devdoc/watchdog.md

```diff
@@ -9,9 +9,10 @@ This may be used to detect code that is stuck in a long-running operation.
 
 Individual watchdog instances can be started with specific timeout times in milliseconds and a callback to call if the watchdog expires. The watchdog instances should be stopped by the caller when the task they are watching completes so that no callback is fired.
 
-Each time a watchdog timer instances is started, it has the following state:
- - `RUNNING` : The timer has started and if it expires it will call the callback.
- - `STOP` : The explicit stop was called and if the timer expires then nothing will happen.
+The watchdog timer uses state management to operate:
+- On start, it will move to the open state on success.
+- callbacks will only be called in the open state. The callback will call `sm_exec_begin` to ensure timer is ready and will call `sm_exec_end` upon completion.
+- When timer is reset or stopped, the state is transistioned to closed first to ensure any executing callback is complete.

```

Comment:
Nice that we explain this. I would suggest going even further a bit and explaining what happens to sm in stop and reset separately.